### PR TITLE
sigsum-go 0.9.1 (new formula)

### PR DESCRIPTION
**Pull request by printfn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/sigsum.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    system "go", "build", *std_go_args(output: bin/"sigsum-submit", ldflags: "-s -w"), "./cmd/sigsum-submit"
+    system "go", "build", *std_go_args(output: bin/"sigsum-token", ldflags: "-s -w"), "./cmd/sigsum-token"
+    system "go", "build", *std_go_args(output: bin/"sigsum-verify", ldflags: "-s -w"), "./cmd/sigsum-verify"
+    system "go", "build", *std_go_args(output: bin/"sigsum-witness", ldflags: "-s -w"), "./cmd/sigsum-witness"
+  end
+
+  test do
+    system bin/"sigsum-key", "gen", "-o", "key-file"
+    shell_output("cat #{bin}/sigsum-key | #{bin}/sigsum-key sign -k key-file -o signature")
+    shell_output("cat #{bin}/sigsum-key | #{bin}/sigsum-key verify -k key-file.pub -s signature")
```
Comment (chenrui333) [inline Formula/s/sigsum.rb:23]:
```suggestion
    pipe_output("#{bin}/sigsum-key sign -k key-file -o signature", "#{bin}/sigsum-key")
    pipe_output("#{bin}/sigsum-key verify -k key-file.pub -s signature", "#{bin}/sigsum-key")
```

File: Formula/s/sigsum.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(output: bin/"sigsum-key", ldflags: "-s -w"), "./cmd/sigsum-key"
+    system "go", "build", *std_go_args(output: bin/"sigsum-monitor", ldflags: "-s -w"), "./cmd/sigsum-monitor"
+    system "go", "build", *std_go_args(output: bin/"sigsum-submit", ldflags: "-s -w"), "./cmd/sigsum-submit"
+    system "go", "build", *std_go_args(output: bin/"sigsum-token", ldflags: "-s -w"), "./cmd/sigsum-token"
+    system "go", "build", *std_go_args(output: bin/"sigsum-verify", ldflags: "-s -w"), "./cmd/sigsum-verify"
+    system "go", "build", *std_go_args(output: bin/"sigsum-witness", ldflags: "-s -w"), "./cmd/sigsum-witness"
```
Comment (chenrui333) [inline Formula/s/sigsum.rb:17]:
```suggestion%w[
    %w[
      sigsum-key
      sigsum-monitor
      sigsum-submit
      sigsum-token
      sigsum-verify
      sigsum-witness
    ].each do |cmd|
      system "go", "build", *std_go_args(output: bin/cmd, ldflags: "-s -w"), "./cmd/#{cmd}"
    end
end
```

Comment (chenrui333) [inline Formula/s/sigsum.rb:17]:
```suggestion
    %w[
      sigsum-key
      sigsum-monitor
      sigsum-submit
      sigsum-token
      sigsum-verify
      sigsum-witness
    ].each do |cmd|
      system "go", "build", *std_go_args(output: bin/cmd, ldflags: "-s -w"), "./cmd/#{cmd}"
    end
```

File: Formula/s/sigsum.rb (hunk 1 above)
Comment (printfn) [reply Formula/s/sigsum.rb]: done

File: Formula/s/sigsum.rb (hunk 2 above)
Comment (printfn) [reply Formula/s/sigsum.rb]: done

File: Formula/s/sigsum.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class Sigsum < Formula
+  desc "Key transparency toolkit"
+  homepage "https://sigsum.org"
+  url "https://git.glasklar.is/sigsum/core/sigsum-go.git",
+    tag:      "v0.9.1",
+    revision: "a7677ede2a545f7b53bc07576fb299ed402859bc"
```
Comment (chenrui333) [inline Formula/s/sigsum.rb:6]:
```suggestion
  url "https://git.glasklar.is/sigsum/core/sigsum-go/-/archive/v0.9.1/sigsum-go-v0.9.1.tar.bz2"
  sha256 "68f3a66af6727fa1b44d7dd2429748b23b8767547750113318494c58d7b04ae8"
```

Comment (printfn) [reply Formula/s/sigsum.rb]: done

File: Formula/s/sigsum.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      sigsum-witness
+    ].each do |cmd|
+      system "go", "build", *std_go_args(output: bin/cmd, ldflags: "-s -w"), "./cmd/#{cmd}"
+    end
+  end
+
+  test do
+    system bin/"sigsum-key", "gen", "-o", "key-file"
+    pipe_output("#{bin}/sigsum-key sign -k key-file -o signature", "#{bin}/sigsum-key")
+    pipe_output("#{bin}/sigsum-key verify -k key-file.pub -s signature", "#{bin}/sigsum-key")
```
Comment (chenrui333) [inline Formula/s/sigsum.rb:26]: can we also verify the output dump?

```suggestion
    pipe_output("#{bin}/sigsum-key sign -k key-file -o signature", bin/"sigsum-key")
    pipe_output("#{bin}/sigsum-key verify -k key-file.pub -s signature", bin/"sigsum-key")
```

File: Formula/s/sigsum.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class Sigsum < Formula
```
Comment (chenrui333) [inline Formula/s/sigsum.rb:1]: maybe do as sigsum-go, that is what other distributions are doing, https://repology.org/project/sigsum-go/versions. what do you think?

```suggestion
class SigsumGo < Formula
```

File: Formula/s/sigsum.rb (hunk 4 above)
Comment (printfn) [reply Formula/s/sigsum.rb]: I'm not sure what you mean by verifying the output dump? This test case verifies the signature against the original file, not sure what else it should be verifying?

File: Formula/s/sigsum.rb (hunk 5 above)
Comment (printfn) [reply Formula/s/sigsum.rb]: makes sense, I've renamed it

**Review by chenrui333 — APPROVED** [review]
Thanks for the great work!

**Review by ZhongRuoyu — APPROVED** [review]
Thanks @printfn @chenrui333!

**Pull request merged**